.github/workflows/invalidatePRs.sh

```diff
@@ -39,19 +35,58 @@ function log() {
   fi
 }
 
+declare -A listSHA
+
+# Get head sha on master branch
+masterCommit=$(curl -s \
+  -H "Accept: application/vnd.github.v3+json" \
+  -H "Authorization: token $oauth" \
+  https://api.github.com/repos/imodeljs/imodeljs/commits/master)
+
+masterHeadCommit=$(echo ${masterCommit} | jq -r '. | @base64')
+masterSha=$(_jq  ${masterHeadCommit} '.sha')
+
+listSHA["master"]=${masterSha}
+echo "Master SHA is ${listSHA["master"]}"
+
 for pr in $(echo "${prs}" | jq -r '.[] | @base64'); do
-  #echo ${pr} | base64 --decode
-  log "$(_jq '.title') (#$(_jq '.number'))"
+  log "$(_jq ${pr} '.title') (#$(_jq ${pr} '.number'))"
+
+  if [[ "open" != $(_jq ${pr} '.state') ]] || [[ "true" == $(_jq ${pr} '.draft') ]]; then
+    log "  Skipping due to state=$(_jq ${pr} '.state') and draft=$(_jq ${pr} '.draft')."
+    continue
+  fi
+
+  ref=$(_jq ${pr} '.base.ref')
+  sha=$(_jq ${pr} '.base.sha')
 
-  if [[ "open" != $(_jq '.state') ]] || [[ "true" == $(_jq '.draft') ]]; then
-    log "  Skipping due to state=$(_jq '.state') and draft=$(_jq '.draft')."
+  ShaToCompare=${listSHA[${ref}]}
+
+  # Keep the list of branch head sha
+  if [[ $ShaToCompare == "" ]]
+  then
+    refCommit=$(curl -s \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H "Authorization: token $oauth" \
+              https://api.github.com/repos/imodeljs/imodeljs/commits/${ref})
+    refHeadCommit=$(echo ${refCommit} | jq -r '. | @base64')
+    refHeadSha=$(_jq  ${refHeadCommit} '.sha')
+    listSHA[${ref}]=${refHeadSha}
+    ShaToCompare=$refHeadSha
```

Comment:
nit: this would be more readable if the variable name casing was consistent (i.e., rename this to `shaToCompare`)